src/Core/Notifications/INotificationService.cs

```diff
@@ -56,29 +56,17 @@ public void ShowNotification(INotification notification)
             {
                 throw new ArgumentNullException(nameof(notification));
             }
-            // todo: check if already shown
+
+            if (activeNotification?.Item2.Id == notification.Id)
```

Comment:
Minor optimization: the "analysis failed" notification can be triggered multiple times, there is no reason to block the UI thread each time.

Reply:
It makes sense for this notification but can we have a scenario where we need to trigger it multiple scenario? (I can't think of any)
Also instead of using Item2 maybe we should use field names for the tuple to increase readibility.

Reply:
As mentioned [in the other PR](https://github.com/SonarSource/sonarlint-visualstudio/pull/3199#discussion_r951201269), there is a more restrictive check that makes this one redundant ("_Only one instance of the notification should be shown per VS session_").